pandas/io/json/_json.py

```diff
@@ -1211,23 +1211,25 @@ def _try_convert_data(
             if result:
                 return new_data, True
 
+        converted = False
         if self.dtype_backend is not lib.no_default and not is_axis:
             # Fall through for conversion later on
             return data, True
         elif is_string_dtype(data.dtype):
             # try float
             try:
                 data = data.astype("float64")
+                converted = True
             except (TypeError, ValueError):
                 pass
 
-        if data.dtype.kind == "f":
-            if data.dtype != "float64":
-                # coerce floats to 64
-                try:
-                    data = data.astype("float64")
-                except (TypeError, ValueError):
-                    pass
+        if data.dtype.kind == "f" and data.dtype != "float64":
```

Comment:
Do you know where float32 values are coming from in the first place? I am surprised this is necessary

Reply:
Same...I'm guessing this is just an overly safe check but just leaving this unchanged since this is a regression fix